File: frontend/src/services/subCategoryService.js
import api from './api';

const subCategoryService = {
    getAllSubCategories: async () => {
        const response = await api.get('/subcategories');
        return response.data;
    },
    getSubCategoryById: async (id) => {
        const response = await api.get(`/subcategories/${id}`);
        return response.data;
    },
    getSubCategoriesByCategory: async (categoryId) => {
        const response = await api.get(`/subcategories/category/${categoryId}`);
        return response.data;
    },
    createSubCategory: async (data) => {
        const response = await api.post('/subcategories', data);
        return response.data;
    },
    updateSubCategory: async (id, data) => {
        const response = await api.put(`/subcategories/${id}`, data);
        return response.data;
    },
    deleteSubCategory: async (id) => {
        await api.delete(`/subcategories/${id}`);
    },
};

export default subCategoryService;

File: src/main/java/com/ecommerce/service/SubCategoryService.java
package com.ecommerce.service;

import com.ecommerce.dto.SubCategoryCreateDTO;
import com.ecommerce.dto.SubCategoryDTO;

import java.util.List;

public interface SubCategoryService {
    SubCategoryDTO createSubCategory(SubCategoryCreateDTO dto);
    SubCategoryDTO getSubCategoryById(Long id);
    List<SubCategoryDTO> getAllSubCategories();
    List<SubCategoryDTO> getSubCategoriesByCategoryId(Long categoryId);
    SubCategoryDTO updateSubCategory(Long id, SubCategoryCreateDTO dto);
    void deleteSubCategory(Long id);
}